### PR TITLE
Use MultipartFormData from StreamCore

### PR DESCRIPTION
**Pull request by laevandus**
### 🔗 Issue Links

- https://linear.app/stream/issue/IOS-1905/openapi-multipartformdata-to-streamcore
- Depends on https://github.com/GetStream/stream-core-swift/pull/68

### 🎯 Goal

Move the shared multipart form-data encoder into StreamCore so StreamChat consumes the common implementation.

### 📝 Summary

- Replaced StreamChat's duplicate MultipartFormData implementation with StreamCore.MultipartFormData.
- Updated multipart request headers to use the selected instance boundary.
- Configured the Xcode project to resolve StreamCore's multipart-form-data branch only.
- Updated the WebSocket ping-controller mock to match the current StreamCore factory signature.
- Left Package.swift and Package.resolved unchanged.

### 🛠 Implementation

StreamChat imports StreamCore for CDN uploads and encodes request bodies with MultipartFormData.encode(). The Content-Type header reads multipartFormData.boundary so it always matches the encoded body. The Xcode project uses the upstream branch until StreamCore PR #68 is merged and released. The test mock now accepts all four createPingController factory arguments required by StreamCore.

### 🎨 Showcase

Not applicable — internal SDK change with no visual behavior.

| Before | After |
| ------ | ----- |
| Local multipart encoder | StreamCore multipart encoder |

### 🧪 Manual Testing Notes

- Focused StreamCore MultipartFormData Swift Testing suite passed.
- StreamAttachments production build passed.
- StreamChat production build passed while resolving the remote multipart-form-data branch.
- Full StreamChatTests suite passed after updating the WebSocket ping-controller mock.

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] Changelog is updated with new localization keys
- [x] New code is covered by unit tests
- [ ] Documentation has been updated in the docs-content repo

## Summary by CodeRabbit

* **Improvements**
  * Updated multipart form-data request encoding for POST, PATCH, and PUT requests.
  * Improved boundary handling so `Content-Type` headers match the generated multipart body.
* **Tests**
  * Revised multipart request tests to validate the new encoding and boundary values.
* **Chores**
  * Updated the minimum required version for the `stream-core-swift` dependency to improve package resolution consistency.

### CODE REVIEW

File: StreamChat.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -5453,8 +5453,8 @@
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/GetStream/stream-core-swift";
 			requirement = {
-				kind = upToNextMajorVersion;
-				minimumVersion = 0.9.0;
+				branch = multipart-form-data;
+				kind = branch;
```
[the remark this answers is not in the document]
Comment (laevandus) [reply StreamChat.xcodeproj/project.pbxproj]: It is OK, this gets updated before merge.

File: TestTools/StreamChatTestTools/Mocks/StreamChat/WebSocketClient/WebSocketClient_Mock.swift [hunk 2]
```diff
@@ -36,7 +36,7 @@ final class WebSocketClient_Mock: WebSocketClient, @unchecked Sendable {
     ) {
         var environment = WebSocketClient.Environment.mock
         if let pingController = pingController {
-            environment.createPingController = { _, _, _ in pingController }
+            environment.createPingController = { _, _, _, _ in pingController }
```
Comment (laevandus) [inline TestTools/StreamChatTestTools/Mocks/StreamChat/WebSocketClient/WebSocketClient_Mock.swift:39]: Video SDK compatibility related change

File: StreamChat.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -5453,8 +5453,8 @@
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/GetStream/stream-core-swift";
 			requirement = {
-				kind = upToNextMajorVersion;
-				minimumVersion = 0.9.0;
+				branch = multipart-form-data;
```
Comment (martinmitrevski) [inline StreamChat.xcodeproj/project.pbxproj]: reminder for this one

**Review by martinmitrevski — APPROVED** [review]
lgtm
